src/core/main.c

```diff
@@ -2048,6 +1999,65 @@ static int do_queue_default_job(
         return 0;
 }
 
+static void reset_arg_rlimit_nofile(const struct rlimit *saved_rlimit_nofile) {
```

Comment:
maybe call this fallback_rlimit_nofile()? after all it doesn't actually reset anything... it doesn't work at all if there's already something set.

Reply:
get_default_rlimit_nofile() ?

Reply:
augment_default_rlimit_nofile()?

Reply:
I think your first proposition is better, I'll pick it up.